virtual_rainforest/models/soil/soil_model.py

```diff
@@ -132,29 +147,122 @@ def spinup(self) -> None:
         """Placeholder function to spin up the soil model."""
 
     def update(self) -> None:
-        """Placeholder function to update the soil model."""
+        """Function to update the soil model."""
 
-        # Convert time step from seconds to days
-        dt = self.update_interval.astype("timedelta64[s]").astype(float) / (
-            60.0 * 60.0 * 24.0
-        )
-
-        carbon_pool_updates = self.carbon.calculate_soil_carbon_updates(
-            self.data,
-            self.data["pH"],
-            self.data["bulk_density"],
-            self.data["soil_moisture"],
-            self.data["soil_temperature"],
-            self.data["percent_clay"],
-            dt,
-        )
+        # Find carbon pool updates by integration
+        updated_carbon_pools = self.integrate_soil_model()
 
         # Update carbon pools (attributes and data object)
         # n.b. this also updates the data object automatically
-        self.carbon.update_soil_carbon_pools(self.data, carbon_pool_updates)
+        self.replace_soil_pools(updated_carbon_pools)
 
         # Finally increment timing
         self.next_update += self.update_interval
 
     def cleanup(self) -> None:
         """Placeholder function for soil model cleanup."""
+
+    def replace_soil_pools(self, new_pools: Dataset) -> None:
+        """Replace soil pools with previously calculated new pool values.
+
+        The state of particular soil pools will effect the rate of other processes in
+        the soil module. These processes in turn can effect the exchange rates between
+        the original soil pools. Thus, a separate update function is necessary so that
+        the new values for all soil module components can be calculated on a single
+        state, which is then updated (using this function) when all values have been
+        calculated.
+
+        Args:
+            new_pools: Array of new pool values to insert into the data object
+        """
+
+        self.data["low_molecular_weight_c"] = new_pools["new_lmwc"]
+        self.data["mineral_associated_om"] = new_pools["new_maom"]
+
+    def integrate_soil_model(self) -> Dataset:
+        """Function to integrate the soil model.
```

Comment:
We already know it's a soil model, so I think you can just call this `integrate()`. And, again, you can drop the "Function to..." bit.

```suggestion
    def integrate(self) -> Dataset:
        """Integrate the soil model.
```